src/executables/sysrepo-plugind.c

```diff
@@ -527,6 +534,34 @@ main(int argc, char **argv)
         goto cleanup;
     }
 
+    /* open and lock pid file before plugins are loaded, but don't update it yet 
+     * plugins may fail, and do some initialization that must finish before we are
+     * considered up and running. No need to create a pid file if we're in debug
+     * mode (we're not a daemon)
```

Comment:
nit: In general, I prefer comments at the place where they "match the code". I.e., in this case the explanation why using the debug mode and asking for a PID writeout doesn't make sense should IMHO go to that `SRPLG_LOG_WRN` branch (and it should not just say the same thing as the code, but rather explain *why*, otherwise I suggest to simply remove that comment)

Reply:
Comment(s) updated. I also removed the `close(pidfd)` as it removes the lock!